lib/shopify_cli/theme/dev_server/cdn_assets.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require_relative "cdn/cdn_helper"
+require_relative "local_assets"
+
+module ShopifyCLI
+  module Theme
+    module DevServer
+      class CdnAssets
+        include Cdn::CdnHelper
+
+        ASSETS_PROXY_PATH = "/cdn_asset"
+        ASSETS_CDN = "//cdn.shopify.com"
+        ASSETS_CDN_REGEX = %r{((https?:)?#{ASSETS_CDN}(.*\.(css|js)))}
+
+        def initialize(app, theme:)
+          @app = app
+          @theme = theme
+        end
+
+        def call(env)
+          path = env["PATH_INFO"]
+
+          # Serve assets from CDN
+          return serve_asset(env, path) if path.start_with?(ASSETS_PROXY_PATH)
+
+          # Serve maps from CDN
+          return serve_path(env, path) if path.end_with?(".map")
```

Comment:
Wouldn't this get handled by the line above if you add `map` to the `ASSETS_CDN_REGEX` pattern?

Reply:
Excellent suggestion, @macournoyer 🚀 I've introduced a new commit mapping source maps to the `/cdn_asset` prefix as well :)